pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -757,9 +786,88 @@ func (gce *GCECloud) EnsureLoadBalancer(clusterName string, apiService *api.Serv
 
 	status := &api.LoadBalancerStatus{}
 	status.Ingress = []api.LoadBalancerIngress{{IP: ipAddress}}
+
 	return status, nil
 }
 
+func serviceNeedsHealthCheck(service *api.Service) (b bool) {
+	if l, ok := service.Annotations[apiservice.AnnotationOnlyNodeLocalEndpoints]; ok {
+		var err error
+		b, err = strconv.ParseBool(l)
+		if err != nil {
+			glog.Errorf("Failed to parse annotation %v: %v", apiservice.AnnotationOnlyNodeLocalEndpoints, err)
+			return false
+		}
+		return b
+	}
+	return false
+}
+
+func getServiceHealthCheckNodePort(service *api.Service) int32 {
+	if serviceNeedsHealthCheck(service) {
```

Comment:
nit: can you do a !serviceNeedsHealthCheck and return early to avoid the extra indet?